site/_posts/2017-09-29-miqldap-to-sssd.md

```diff
@@ -0,0 +1,305 @@
+---
+
+How does one user the *miqldap_to_sssd* conversion tool?
+
+# MiqLdap To SSSD Conversion Tool
+
+**Contents:**
+---------------------------------------------------------------------
+
+- **A brief Why before The How:**
+- **The How**
+- **Let's kick the tires**
+  + **Example 1:** The simple Case:
+  + **Example 2:** A case with a small complication:
+  + **Example 3:** A caser with a different small complication:
+- **The other miqldap_to_sssd arguments**
+- **Debugging Failed Conversions**
+
+**A brief Why before The How:**
+---------------------------------------------------------------------
+
+Why did the *miqldap_to_sssd* conversion tool come to be?
+
+MiQ is delivered with the built in LDAP client, *MiqLdap*. Maintaining our own LDAP
+client is unnecessary as there are multiple mechanisms built into the operating system that provide an interface to an LDAP directory.
+
+**The How**
+---------------------------------------------------------------------
+
+The *miqldap_to_sssd* conversion tool will convert an MiQ appliance configured to user the our *MiqLdap* client to using external authentication.
+
+To perform this re-configuration *miqldap_to_sssd* simply automates the instructions for manually configuring an MiQ Appliance’s external authentication to work against LDAP [(as documented here)](http://manageiq.org/docs/reference/latest/auth/ldap). It then customizes the configuration to emulate the *MiqLdap* configuration settings. 
+
+The *MiqLdap* client is enabled when the appliance authentication mode is set to **LDAP(S)**
+The external authentication mechanism is enabled when the appliance authentication mode is set to **External (httpd)**
+
+
+**Let's kick the tires**
+---------------------------------------------------------------------
+
+I'll present 3 examples:
+
+- **Example 1:** The simple Case:
+  + Authentication *Mode LDAP* and with *Get User Groups from LDAP* **checked**
+- **Example 2:** A case with a small complication:
+  + Authentication *Mode LDAP* but with *Get User Groups from LDAP* **un-checked**
+- **Example 3:** A caser with a different small complication:
+  + Authentication *Mode LDAPS* using certificate files 
+
+**Example 1:** The simple Case:
+---------------------------------------------------------------------
+
+Prior to running the *miqldap_to_sssd* conversion the appliance is configured with:
+
+Authentication *Mode LDAP* and with *Get User Groups from LDAP* **checked**
+
+Authenticaiton Configuration Settings:
+
+|                               |                 |
+|:-----------------------------:|:----------------|
+| *Mode:*                       | **LDAP**        |
+| *Get User Groups from LDAP:*  | **checked**     |
+| *Get Roles from Home Forest:* | **checked**     |
+| *Base DN:*                    | **filled in**   |
+| *Bind DN:*                    | **filled in**   |
+| *Bind Password:*              | **filled in**   |
+
+
+Screenshot showing the appliance configuration for Mode: LDAP:
```

Comment:
Not sure if intended, but this is showing up as:
Screenshot showing the appliance configuration for Mode: LDAP: Screenshot Showing Appliance Configuration Mode: LDAP

With the second half showing as a link to the image.